### PR TITLE
Support for touch devices

### LINKED ISSUE #2: Support on touch devices
Would that be possible? Maybe relying on `onTouchStart/End` or using https://github.com/JedWatson/react-tappable?

pd: congrats on the project!

### PR DESCRIPTION
**Pull request by mapreal19**
- Relies on selection change event instead of onmouseup
- Removes contextmenu since it overlaps the highlight popup
- Closes #2

### CODE REVIEW

File: src/components/PdfLoader.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import React, { Component } from "react";
 
 import type { T_PDFJS, T_PDFJS_Document } from "../types";
 
-import { PDFJS } from "pdfjs-dist";
+import pdfjs from 'pdfjs-dist/webpack';
```
Comment (mapreal19) [inline src/components/PdfLoader.js]: `webpack` import needed for the worker

**Comment by mapreal19** [comment]
highlight

**Comment by mapreal19** [comment]
So after updating pdfjs it seems the bindmouse hack is no longer needed

File: package-lock.json [hunk 2]
```diff
@@ -8039,12 +8042,12 @@
       }
     },
     "pdfjs-dist": {
-      "version": "2.0.159",
-      "resolved": "https://unpm.uberinternal.com/pdfjs-dist/-/pdfjs-dist-2.0.159.tgz",
```
Comment (mapreal19) [inline package-lock.json]: why uberinteral? What about using yarn instead?

**Comment by mapreal19** [comment]
@agentcooper oh so it seems the state got stuck with MouseDown set to true. Now it seems to work. Also added debounce, not sure if fine but I think it's great for not calling `onMouseUp` after every selectChange (which could be 50+ times if picking a large paragraph)

File: package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
     "prettier": "prettier --write 'src/**/*.js' 'demo/src/**/*.js'"
   },
   "dependencies": {
+    "lodash": "^4.17.10",
```
Comment (agentcooper) [inline package.json]: Can you switch to https://www.npmjs.com/package/lodash.debounce?
(I don't know whether tree shaking is there yet).

File: src/components/PdfAnnotator.js [hunk 4]
```diff
@@ -425,7 +427,7 @@ class PdfAnnotator<T_HT: T_Highlight> extends Component<
       range
     });
 
-    this.afterSelection();
+    debounced(this.onMouseUp);
```
Comment (agentcooper) [inline src/components/PdfAnnotator.js]: 👍

File: src/components/PdfAnnotator.js [hunk 5]
```diff
@@ -531,6 +533,7 @@ class PdfAnnotator<T_HT: T_Highlight> extends Component<
     return (
       <div
         ref={node => (this.containerNode = node)}
+        onMouseUp={this.onMouseUp}
```
Comment (agentcooper) [inline src/components/PdfAnnotator.js]: Does this work on touch devices?

File: package.json (hunk 3 above)
Comment (mapreal19) [reply package.json]: using the `cherryPick` option will do? https://github.com/insin/nwb/blob/master/docs/Configuration.md#cherrypick-string--arraystring

File: src/components/PdfAnnotator.js (hunk 5 above)
Comment (mapreal19) [reply src/components/PdfAnnotator.js]: Checked and not really. Just hit a brick wall. I'll try to find a solution later

**Comment by mapreal19** [comment]
highlight

**Comment by mapreal19** [comment]
@agentcooper made few tweaks. What do you think? Had to get rid of the mouseDown state.

Tested on an Android tablet and worked fine

**Comment by agentcooper** [comment]
@mapreal19 sorry for the delay. Everything seems to work now, thanks!

File: src/components/PdfLoader.js (hunk 1 above)
Comment (DenisKrasnov) [reply src/components/PdfLoader.js]: Seems this import doesn’t work with webpack 4.

Comment (mapreal19) [reply src/components/PdfLoader.js]: which error did you get?

Comment (DenisKrasnov) [reply src/components/PdfLoader.js]:
```
DeprecationWarning: Tapable.apply is deprecated. Call apply on the plugin directly instead
```

File: src/components/PdfAnnotator.js (hunk 5 above)
Comment (amitsood) [reply src/components/PdfAnnotator.js]: Hi, Touch seems to be working on android but not on IOS  devices.

**Comment by amitsood** [comment]
@agentcooper, Touch is working on Android only and not on IOS (iphone or iPad). Can you please advise how to get that working

**Pull request merged**